### DISCUSSION TITLE
SyntaxError: Unexpected token 'export' @adonisjs/websocket-client

### DISCUSSION THREAD

**Discussion by JonhnyDev** [vercel/next.js#18566] [Help]
Hi! can you help me?
I am using AdonisJS in my backend, in my current React Web project adonis works normally but passing it to NextJS when installing "yarn add @adonisjs/websocket-client" Link: "https://adonisjs.com/docs/4.1/websocket-client" it gives an error, both on the SSG and SSR pages, the error occurs when trying to compile the code with" yarn build ".

![image](https://user-images.githubusercontent.com/43152652/97759638-3a6ab400-1ae0-11eb-9676-a7b4a39928d4.png)

src/socket.js:

import Ws from '@adonisjs/websocket-client';
export class SocketConnection {
  connect () {
    this.ws = Ws('wss://chatcomposer.com')
    //.withApiToken(token)
    .connect();

    this.ws.on('open', () => {
      console.log('Connection initialized')
    });

    this.ws.on('close', () => {
      console.log('Connection closed')
    });

    return this
  }

  subscribe (channel, handler) {
    if (!this.ws) {
      setTimeout(() => this.subscribe(channel), 1000)
    } else {
      const result = this.ws.subscribe(channel);

      result.on('message', message => {
        console.log('Incoming', message);
        handler(message)
      });

      result.on('error', (error) => {
        console.error(error)
      });

      return result
    }
  }
}

export default new SocketConnection()

Does anybody know how to solve this?
Apparently it's something in Next JS, since in my React and React Native project it compiles and works normally.

In React Web working:
![image](https://user-images.githubusercontent.com/43152652/97762296-d4355f80-1ae6-11eb-9bdd-f168e65793be.png)

**Comment by aligntogether** [comment]
Did you resolve it? I am also facing the same.

**Comment by mfrancis1234** [comment]
Someone with any solution to this problem? I have the same error!

**Reply by JonhnyDev** [reply]
No, this adonis package is full of problems i recommend that you change the adonis / websocket for socket.io

**Reply by mfrancis1234** [reply]
> No, this adonis package is full of problems i recommend that you change the adonis / websocket for socket.io

Hi JonhnyDev, Thank you for your referral

**Reply by Rafazeon** [reply]
@JonhnyDev 
nextjs gives the error because of server side rendering, because when you use adonis websocket in pure react you are only using it in the browser without using nodejs together, to solve the problem you need to disable the ssr of the page that will use the socket example:

```js
import dynamic from "next/dynamic";

const DynamicOrderComponentWithNoSSR = dynamic(
  () => import('~/components/Order'),
  { ssr: false }
)

return <DynamicOrderComponentWithNoSSR />
```

this solves the problem
